pkg/gce-pd-csi-driver/controller.go

```diff
@@ -704,10 +714,12 @@ func (gceCS *GCEControllerServer) executeControllerUnpublishVolume(ctx context.C
 		klog.V(4).Infof("ControllerUnpublishVolume succeeded for disk %v from node %v. Already not attached.", volKey, nodeID)
 		return &csi.ControllerUnpublishVolumeResponse{}, nil
 	}
-
+	diskToUnpublish, _ := gceCS.CloudProvider.GetDisk(ctx, gceCS.CloudProvider.GetDefaultProject(), volKey, gce.GCEAPIVersionV1)
```

Comment:
We should handle the error of this GetDisk call as well, I would say make it not disruptive? and move it to before emit the metrics?

We can do L720:
```
if err != nil {
	diskToUnpublish, getDiskError := gceCS.CloudProvider.GetDisk(ctx, gceCS.CloudProvider.GetDefaultProject(), volKey, gce.GCEAPIVersionV1)

	// If there is an error, log it as a warning only and do not emit the metrics, because diskToUnpublish will be nil

	// Emit metric for error 
	gceCS.Metrics.RecordOperationErrorMetrics("ControllerUnpublishVolume", err, diskToUnpublish.GetPDType())
	return nil, common.LoggedError("unknown detach error: ", err)
}
```

Reply:
Updated error handling